source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -158,7 +158,7 @@ static Api::IoCallResult<FileInfo> infoFromStat(absl::string_view path, const st
 }
 
 static Api::IoCallResult<FileInfo> infoFromStat(absl::string_view path, const struct stat& s) {
-  return infoFromStat(path, s, typeFromStat(s));
+  return infoFromStat(path, s, typeFromStat(s)); // NOLINT(clang-analyzer-cplusplus.NewDeleteLeaks)
```

Comment:
Is this because of the resultFailure call's `IoFileErrorPtr` with custom deleter?

Reply:
That's my assumption, I'm having a go at changing it and running clang-tidy CI right now (but the clang-tidy CI takes *aaages*.)